### PR TITLE
Add scripts to auto-update build tools and submodules

### PR DESCRIPTION
**Pull request by natemcmaster**
Part of https://github.com/aspnet/Universe/issues/577

These scripts will run on CI and update to new versions of submodules and build tools.

CI flow:

1. Run the ./script/Update*.ps1 script and save changes to local copy
2. Run some kind of test to ensure things still build. Examples: run the compilation step, ensure compile passes, etc.
3. If the test passes, `git push` updates

### CODE REVIEW

File: scripts/UpdateBuildTools.ps1 [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    return $version
+}
+
+Push-Location $RepoRoot
+try {
+    Ensure-Git
+
+    $oldVersion = Get-KoreBuildVersion
+
+    & "$RepoRoot/run.ps1" -Update default-build '/t:Cow'
```
Comment (ryanbrandenburg) [inline scripts/UpdateBuildTools.ps1]: Isn't running default-build with `/t:Cow` the same as running `install-tools`? If not I feel like it's worth clarifying what all we expect this to do.

File: scripts/UpdateBuildTools.ps1 [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    | % {
+        if ($_ -like '*<InternalAspNetCoreSdkPackageVersion>*') {
+            "    <InternalAspNetCoreSdkPackageVersion>$newVersion</InternalAspNetCoreSdkPackageVersion>"
+        } else {
+            $_
+        }
+    }
+    $deps | Set-Content -Encoding UTF8 "$RepoRoot/build/dependencies.props"
+
+    & git add "$RepoRoot/korebuild-lock.txt"
```
Comment (ryanbrandenburg) [inline scripts/UpdateBuildTools.ps1]: Ah, I see, having accepted our fates to have to update `build/dependencies.props` constantly we're also now committing `korebuild-lock.txt`.

File: scripts/UpdateSubmodules.ps1 [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        }
+        finally {
+            Pop-Location
+        }
+    } `
+    | ? { $_.changed } `
+    | % { "$($_.module) to $($_.newCommit.Substring(0, 8))" }
+
+    if ($changes) {
+        $message = "Updating submodule $( $changes -join ' ,' )"
```
Comment (ryanbrandenburg) [inline scripts/UpdateSubmodules.ps1]: `submodule(s)`?

File: scripts/UpdateBuildTools.ps1 [hunk 4]
```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env powershell
```
Comment (ryanbrandenburg) [inline scripts/UpdateBuildTools.ps1]: We expect this script and the one below to mostly only be run by the CI, correct?

Comment (natemcmaster) [reply scripts/UpdateBuildTools.ps1]: Yes

File: scripts/UpdateBuildTools.ps1 (hunk 1 above)
Comment (natemcmaster) [reply scripts/UpdateBuildTools.ps1]: Yeah. I just need some kind of no-op command so the korebuild-lock.txt file gets updated. I'll add a comment.

File: scripts/UpdateBuildTools.ps1 (hunk 2 above)
Comment (natemcmaster) [reply scripts/UpdateBuildTools.ps1]: Yup. #575 requires it.

**Pull request merged**